src/components/common/DropdownButton.js

```diff
@@ -3,80 +3,45 @@ import React, { useState } from 'react';
 import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
 
 /** --------Props---------------
- * children: array of strings, each string being the name of a Program
+ * children: array of dropDownButtons, each containing the button to switch to another language
  * displayValue: string to be displayed as the placeholder for the dropdown
- * onSelect: function called when an item is selected in the dropdown
+ * toggleProps: props that dictate the styling of dropDownToggle
+ * icon: contains optional icon for editor dropDown
+ * displayClass: contains different classes for editor and sketch dropdownbutton
  */
 /** --------Optional props--------
  * defaultOpen: boolean determining if the dropdown should start off open or closed
  */
 
 const DropdownButton = (props) => {
-  const {
-    dirty,
-    onSelect,
-    children,
-    defaultOpen,
-    displayValue,
-    displayClass,
-    toggleClass,
-    toggleColor,
-    toggleSize,
-  } = props;
+  const { children, icon, defaultOpen, displayValue, displayClass, ...toggleProps } = props;
```

Comment:
Instead of taking the rest of the props as the toggleProps, I think that it should be explicitly passed in as an object toggleprops, and then spread from that.